File: uikit/src/main/java/io/agora/chat/uikit/widget/chatextend/PageDecorationLastJudge.java
package io.agora.chat.uikit.widget.chatextend;

/**
 * Refer to：https://blog.csdn.net/Y_sunny_U/article/details/89500464
 */
public interface PageDecorationLastJudge {
    /**
     * Is the last row in one page
     *
     * @param position
     * @return
     */
    boolean isLastRow(int position);
 
    /**
     * Is the last Colum in one row;
     *
     * @param position
     * @return
     */
    boolean isLastColumn(int position);
 
    boolean isPageLast(int position);
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/common/PageDecorationLastJudge.kt
package io.agora.uikit.common

/**
 * Refer to：https://blog.csdn.net/Y_sunny_U/article/details/89500464
 */
interface PageDecorationLastJudge {
    /**
     * Is the last row in one page
     *
     * @param position
     * @return
     */
    fun isLastRow(position: Int): Boolean

    /**
     * Is the last Colum in one row;
     *
     * @param position
     * @return
     */
    fun isLastColumn(position: Int): Boolean
    fun isPageLast(position: Int): Boolean
}